src/astra_driver.cpp

```diff
@@ -571,41 +576,60 @@ void AstraDriver::newDepthFrameCallback(sensor_msgs::ImagePtr image)
   }
 }
 
-// Methods to get calibration parameters for the various cameras
-sensor_msgs::CameraInfoPtr AstraDriver::getDefaultCameraInfo(int width, int height, double f) const
+// // Methods to get calibration parameters for the various cameras
+sensor_msgs::CameraInfoPtr AstraDriver::getDefaultCameraInfo(int width, int height, bool is_ir_camera) const
 {
+  // left parameters are for IR camera, right parameters are for RBG camera
   sensor_msgs::CameraInfoPtr info = boost::make_shared<sensor_msgs::CameraInfo>();
+  OBCameraParams camera_params = device_->getCameraParameters();
 
-  info->width  = width;
+  info->width = width;
   info->height = height;
 
-  // No distortion
-  info->D.resize(5, 0.0);
   info->distortion_model = sensor_msgs::distortion_models::PLUMB_BOB;
-
-  // Simple camera matrix: square pixels (fx = fy), principal point at center
+  info->D.resize(5);
+  
   info->K.assign(0.0);
-  info->K[0] = info->K[4] = f;
-  info->K[2] = (width / 2) - 0.5;
-  // Aspect ratio for the camera center on Astra (and other devices?) is 4/3
-  // This formula keeps the principal point the same in VGA and SXGA modes
-  info->K[5] = (width * (3./8.)) - 0.5;
   info->K[8] = 1.0;
 
-  // No separate rectified image plane, so R = I
-  info->R.assign(0.0);
-  info->R[0] = info->R[4] = info->R[8] = 1.0;
+  if (is_ir_camera) {
+    info->D.assign(camera_params.l_k, camera_params.l_k + 5);
```

Comment:
The depth camera uses the IR camera distortion and K matrix values.

Reply:
correct